### PR TITLE
HCMPRE-218 adding validation for plan config name 

### PR DESCRIPTION
**Pull request by Priyanka-eGov**
## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced name validation mechanisms, including new error codes and messages for scenarios involving empty or failed name validation.
	- Added methods for regex validation of plan configuration names, improving compliance with naming conventions.
	- Incorporated validation logic related to naming conventions within data handling processes.
  
- **Bug Fixes**
	- Improved error handling and validation processes to enhance data integrity and robustness. 

- **Style**
	- Updated variable naming conventions for improved readability and consistency in the codebase.

### CODE REVIEW

**Comment by Priyanka-eGov** [comment]
@coderabbitai review

File: health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java [hunk 1]
```diff
@@ -48,6 +50,54 @@ public void validateCreate(PlanConfigurationRequest request) {
...
+     * @param request  the plan configuration request containing the plan configuration details
+     * @param mdmsData the MDMS data containing the name validation regex patterns
+     * @throws CustomException if the JSONPath evaluation fails, the name validation list from MDMS is empty,
+     *                         or the plan configuration name validation fails.
+     */
+    public void validatePlanConfigName(PlanConfigurationRequest request, Object mdmsData)
+    {
+        PlanConfiguration planConfiguration = request.getPlanConfiguration();
+
+        final String jsonPathForNameValidation = "$." + MDMS_PLAN_MODULE_NAME + "." + MDMS_MASTER_NAME_VALIDATION + "[*].data";
```
Comment (shashwat-egov) [inline health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java:69]: Move $, . into constants

**Review by shashwat-egov — CHANGES_REQUESTED** [review]

File: health-services/plan-service/src/main/java/digit/config/ServiceConstants.java [hunk 2]
```diff
@@ -84,6 +91,9 @@ public class ServiceConstants {
     public static final String MDMS_MASTER_SCHEMAS = "Schemas";
     public static final String MDMS_MASTER_METRIC = "Metric";
     public static final String MDMS_MASTER_UOM = "Uom";
+    public static final String MDMS_MASTER_NAME_VALIDATION= "MicroplanNamingRegx";
```
Comment (shashwat-egov) [inline health-services/plan-service/src/main/java/digit/config/ServiceConstants.java:94]: get this master name changed with Nipun

File: health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java [hunk 3]
```diff
@@ -48,6 +50,54 @@ public void validateCreate(PlanConfigurationRequest request) {
...
+
+        List<Object> nameValidationListFromMDMS = null;
+        try {
+            nameValidationListFromMDMS = JsonPath.read(mdmsData, jsonPathForNameValidation);
+        } catch (Exception e) {
+            log.error(jsonPathForNameValidation);
+            throw new CustomException(JSONPATH_ERROR_CODE, JSONPATH_ERROR_MESSAGE);
+        }
+
+        if (nameValidationListFromMDMS == null || CollectionUtils.isEmpty(nameValidationListFromMDMS)) {
```
Comment (shashwat-egov) [inline health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java:79]: remove redundant null check

File: health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java [hunk 4]
```diff
@@ -48,6 +50,54 @@ public void validateCreate(PlanConfigurationRequest request) {
...
+    }
+
+    /**
+     * Validates the given input string against the provided regex pattern.
+     *
+     * @param patternString the regex pattern to validate against
+     * @param inputString   the input string to be validated
+     * @return true if the input string matches the regex pattern, false otherwise
+     */
+    public static boolean validateStringAgainstRegex(String patternString, String inputString) {
```
Comment (shashwat-egov) [inline health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java:97]: move this method into a generic util class

Comment (shashwat-egov) [reply health-services/plan-service/src/main/java/digit/service/validator/PlanConfigurationValidator.java:97]: use boxed class as return type

**Review by shashwat-egov — DISMISSED** [review]

**Comment by Priyanka-eGov** [comment]
Ticket link - https://digit-discuss.atlassian.net/browse/HCMPRE-218?
@kavi-egov

**Review by kavi-egov — DISMISSED** [review]

**Review by kavi-egov — APPROVED** [review]

**Pull request merged**